lib_eio_linux/eio_linux.ml

```diff
@@ -1162,11 +1162,90 @@ let net = object
   method getnameinfo = Eio_unix.getnameinfo
 end
 
+let rec wait_for_process flags pid = 
+  try Unix.waitpid flags pid with
+  | Unix.Unix_error (Unix.EINTR, _, _) -> wait_for_process flags pid
+
+external sig_to_host_sig : int -> int = "caml_signal_to_posix_signal"
+
+let pid_to_process close pid = object
+  inherit Eio.Process.t
+  method pid = pid
+  val mutable needs_close = true
+  method needs_close = needs_close
+  method set_needs_close b = needs_close <- b
+  method status =
+    let p, r = Promise.create () in
```

Comment:
You need to create the promise (and do the wait) outside of `status`, which should just await the promise that already exists. Otherwise we'll try to reap the child once for each time the user calls `status`, whereas we actually need to do it exactly once.

Reply:
Yes! Thanks, done in https://github.com/ocaml-multicore/eio/pull/330/commits/4a37efbf581251ad8021f1252509230211ba58c0

Reply:
That only creates the promise outside. It's still doing the wait once for each call. It should just be:
```ocaml
method await_exit = Promise.await exit_status
```